### PR TITLE
support check and repair invalid name 

### PR DESCRIPTION
**Pull request by YuezhangMo**
The updates of this pull request.
1. Fix argument validation check
2. Fix the bug that valid characters were mistakenly detected as invalid characters
3. Add check for invalid characters before setting volume label
4. Add support for repairing invalid characters in fsck.exfat

### CODE REVIEW

**Comment by namjaejeon** [comment]
@hclee Please check it.

File: fsck/fsck.c [hunk 1]
```diff
@@ -685,6 +685,18 @@ static int check_name_dentry_set(struct exfat_de_iter *iter,
...
 
+	ret = exfat_check_name(inode->name, stream_de->stream_name_len);
+	if (ret != stream_de->stream_name_len) {
+		char err_msg[36];
+
+		snprintf(err_msg, sizeof(err_msg),
+			"invalid character '%c' in file name",
+			le16_to_cpu(inode->name[ret]));
+
+		return exfat_repair_rename_ask(&exfat_fsck, iter, inode->name,
```
Comment (hclee) [inline fsck/fsck.c:696]: The error message below is a little confused. Can you append 'has', 'filename' or something else?

```
ERROR: '*' invalid character '*' in file name.
```

Comment (YuezhangMo) [reply fsck/fsck.c:696]: I updated the error message to

```
ERROR: '*' filename has invalid character '*'.
```

Comment (hclee) [reply fsck/fsck.c:696]: Thank you.
Looks good to me.

**Comment by namjaejeon** [comment]
Applied it to #exfat-next.
Thanks!

**Pull request closed without merging**